File: TEK3/DOTNET/DOT_AREA_2017/Modules/action/bourse/CheckBourse.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Net;   

namespace Modules.action.bourse
{
    class CheckBourse : Action
    {
        WebClient client = new WebClient();
        Dictionary<string, string> fields = new Dictionary<string, string>();
        Reaction r = null;
        List<string> content = null;
        float price = 0;

        private float getPriceValue()
        {
            int i = 0;

            content = new List<string>();
            while (++i < 8)
            {
                try
                {
                    content.Add(client.DownloadString("http://www.boursorama.com/bourse/actions/cours_az.phtml?page=" + i));
                }
                catch (Exception ex)
                {
                    Console.WriteLine(ex);
                }
            }

            string number = "";
            string cotation = "class=\"cotation\">";
            Boolean check = false;
            foreach (string cont in content)
            {
                for (int j = 0; j < cont.Length; j++)
                {
                    if (j + fields["business"].Length <= cont.Length && cont.IndexOf(fields["business"], j, fields["business"].Length) == j)
                    {
                        while (j++ < cont.Length && !check)
                            if (j + cotation.Length <= cont.Length && cont.IndexOf(cotation, j, cotation.Length) == j)
                            {
                                j += cotation.Length;
                                for (int k = 0; k < 5; k++)
                                    number += cont[k + j];
                                check = true;
                            }
                    }
                }
            }

            float value = float.Parse(number, CultureInfo.InvariantCulture);
            return (value);
        }

        public List<string> getFields()
        {
            return (new List<string>() { { "business" },
                { "rate" } });
        }

        public void setField(Dictionary<string, string> fields)
        {
            this.fields = fields;
            this.price = getPriceValue();
        }

        public void addReaction(Reaction r)
        {
            this.r = r;
        }

        public void update()
        {
            if (this.r == null)
                return;

            float value = getPriceValue();

            string message = "";
            if ((price - value > (float.Parse(this.fields["rate"]) * -1)) ||
                    (value - price) > float.Parse(this.fields["rate"]))
            {
                price = value;
                message = this.fields["business"] + " " + price;
            }

            if (message != "")
                r.execute();
        }
    }
}

File: TEK3/Java/Java_area_2017/src/main/java/com/area/modules/action/bourse/CheckBourse.java
package com.area.modules.action.bourse;

import com.area.modules.Action;
import com.area.modules.Reaction;

import java.net.URL;
import java.net.URLConnection;
import java.util.*;

public class CheckBourse implements Action {
    Map<String, String> fields = new HashMap<String, String>();
    Reaction r = null;
    List<String> content = null;
    URLConnection connection = null;
    float price;

    public List<String> getFields() {
        return (new ArrayList<String>() {{
            add("business");
            add("rate");
        }});
    }

    public void setField(Map<String, String> fields) {
        this.fields = fields;
    }

    public void addReaction(Reaction r) {
        this.r = r;
    }

    public void update() {
        if (this.r == null)
            return;

        int i = 0;

        content = new ArrayList<>();
        while (++i < 8) {
            try {
                connection = new URL("http://www.boursorama.com/bourse/actions/cours_az.phtml?page=" + i).openConnection();
                Scanner scanner = new Scanner(connection.getInputStream());
                scanner.useDelimiter("\\Z");
                content.add(scanner.next());
            } catch (Exception ex) {
                ex.printStackTrace();
            }
        }

        String number = "";
        String cotation = "class=\"cotation\">";
        boolean check = false;
        for (String cont : content) {
            for (int j = 0; j < cont.length(); j++) {
                if (cont.regionMatches(j, this.fields.get("business"), 0, this.fields.get("business").length())) {
                    while (j++ < cont.length() && !check)
                        if (cont.regionMatches(j, cotation, 0, cotation.length())) {
                            j += cotation.length();
                            for (int k = 0; k < 5; k++)
                                number += cont.charAt(k + j);
                            check = true;
                        }
                }
            }
        }

        float value = Float.parseFloat(number);
        System.out.println(value);
        if ((price - value > (Integer.parseInt(this.fields.get("rate")) * -1)) ||
                (value - price) > Integer.parseInt(this.fields.get("rate"))) {
            price = value;
            r.execute();
        }
    }
}